motor/discovery/gcp/mql_asset_objects.go

```diff
@@ -47,49 +45,49 @@ func getTitleFamily(o gcpObject) (gcpObjectPlatformInfo, error) {
 
 func computeInstances(m *MqlDiscovery, project string, tc *providers.Config, sfn common.QuerySecretFn) ([]*asset.Asset, error) {
 	assets := []*asset.Asset{}
-
-	instances, err := m.GetList("return gcp.project.compute.instances { id name labels zone { name } status networkInterfaces disks { guestOsFeatures } }")
-	if err != nil {
-		return nil, err
+	type instanceDisk struct {
+		GuestOsFeatures []string
+	}
+	type instance struct {
+		Id                string
+		Name              string
+		Labels            map[string]string
+		Zone              struct{ Name string }
+		Status            string
+		NetworkInterfaces []compute.NetworkInterface
+		Disks             []instanceDisk
 	}
-	for i := range instances {
-		b := instances[i].(map[string]interface{})
-		id := b["id"].(string)
-		name := b["name"].(string)
 
-		disks := b["disks"].([]interface{})
-		if disksContainWindows(disks) {
-			log.Debug().Msgf("skipping windows instance %s", name)
-			continue
+	disksContainWindows := func(disks []instanceDisk) bool {
+		for _, d := range disks {
+			for _, f := range d.GuestOsFeatures {
+				if f == "WINDOWS" {
+					return true
+				}
+			}
 		}
+		return false
+	}
 
-		tags := b["labels"].(map[string]interface{})
-		zone := b["zone"].(map[string]interface{})
-		zoneName := zone["name"].(string)
-		status := b["status"].(string)
-		stringLabels := make(map[string]string)
-		for k, v := range tags {
-			stringLabels[k] = v.(string)
-		}
-		stringLabels["mondoo.com/instance"] = id
+	instances, err := GetList[instance](m, "return gcp.project.compute.instances { id name labels zone { name } status networkInterfaces disks { guestOsFeatures } }")
+	if err != nil {
+		return nil, err
+	}
 
-		data, err := json.Marshal(b["networkInterfaces"])
-		if err != nil {
-			log.Error().Msgf("failed to marshal network interfaces for gcp compute instance %s", name)
+	for _, i := range instances {
+		if disksContainWindows(i.Disks) {
+			log.Debug().Msgf("skipping windows instance %s", i.Name)
 			continue
 		}
 
-		var networkIfaces []*compute.NetworkInterface
-		if err := json.Unmarshal(data, &networkIfaces); err != nil {
-			log.Error().Msgf("failed to unmarshal network interfaces for gcp compute instance %s", name)
-			continue
-		}
+		stringLabels := i.Labels
+		stringLabels["mondoo.com/instance"] = i.Id
```

Comment:
i know it's not new in this pr, but should this label key be a const?

Reply:
yup, i fixed it in the base branch